src/convert.jl

```diff
@@ -612,6 +612,52 @@ function convert(::Type{Hyperrectangle},
      return Hyperrectangle(low=l, high=h)
 end
 
+"""
+    convert(::Type{CartesianProduct{N, Interval{N}, Interval{N}}},
+            H::AbstractHyperrectangle{N}) where {N<:Real}
+
+Converts a two-dimensional hyperrectangle to the cartesian product of two
+intervals.
+
+### Input
+
+- `CartesianProduct` -- type used for dispatch
+- `H`                -- hyperrectangle
+
+### Output
+
+The cartesian product of two intervals.
+"""
+function convert(::Type{CartesianProduct{N, Interval{N}, Interval{N}}},
+                 H::AbstractHyperrectangle{N}) where {N<:Real}
+    @assert dim(H) == 2 "the hyperrectangle must be two-dimensional to convert it to "
+            "the cartesian product of intervals, but it is $(dim(H))-dimensional"
```

Comment:
```suggestion
            "the cartesian product of two intervals, but it is $(dim(H))-dimensional"
```
Maybe point to `CartesianProductArray` here: `consider converting to 'CartesianProductArray' instead`